src/Features/Core/Portable/Shared/Options/ServiceFeatureOnOffOptions.cs

```diff
@@ -35,5 +35,34 @@ public static bool IsClosedFileDiagnosticsEnabled(OptionSet options, string lang
 
             return option.Value;
         }
+
+        /// <summary>
+        /// Option to disable analyzer execution during live analysis.
+        /// </summary>
+        public static readonly Option<bool> DisableAnalyzers = new Option<bool>(
+            nameof(ServiceFeatureOnOffOptions), nameof(DisableAnalyzers), defaultValue: false,
+            storageLocations: new RoamingProfileStorageLocation($"Options.DisableAnalyzers"));
+
+        /// <summary>
+        /// Option to turn off all background analysis to improve performance.
+        /// </summary>
+        public static readonly Option<bool> PowerSaveMode = new Option<bool>(
+            nameof(ServiceFeatureOnOffOptions), nameof(PowerSaveMode), defaultValue: false,
+            storageLocations: new RoamingProfileStorageLocation($"Options.PowerSaveMode"));
```

Comment:
can you clarify the option.  it is set dynamically when the system goes into low-power mode, or is it set by the user to say "hey, when i go into low power, then don't do as much work".

Reply:
This option is only controlled by the end user. The dynamic low VM mode sets a separate boolean flag (below in this file) that turns on power save mode just for current VS session (not serialized/saved anywhere).

Reply:
so... what i find weird is this.  We're driving this by low VM.  But we're not driving by low-battery (where i would want the power savings)?   Wouldn't the latter make more sense.

Also, by disabling OOP, don't you hurt VM by packing more into the single process?  Or is this machine-wide VM?

Reply:
> so... what i find weird is this. We're driving this by low VM. But we're not driving by low-battery (where i would want the power savings)? Wouldn't the latter make more sense.

We can also do that - I just reused the existing low VM listener that we already have that handles scenarios where we need to suspend features to improve IDE experience. We can also do a similar handling for low battery scenario.

Reply:
Nifty!